pkg/fleet/internal/winregistry/winregistry.go

```diff
@@ -42,3 +43,28 @@ func GetProgramDataDirForProduct(product string) (path string, err error) {
 	path = val
 	return
 }
+
+// GetAgentUserName returns the user name for the Agent, stored in the registry by the Installer MSI
+func GetAgentUserName() (string, error) {
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, "SOFTWARE\\Datadog\\Datadog Installer", registry.QUERY_VALUE)
+	if err != nil {
+		return "", err
+	}
+	defer k.Close()
+
+	user, _, err := k.GetStringValue("installedUser")
+	if err != nil {
+		return "", fmt.Errorf("could not read installedUser in registry: %s", err)
+	}
+
+	domain, _, err := k.GetStringValue("installedDomain")
+	if err != nil {
+		return "", fmt.Errorf("could not read installedDomain in registry: %s", err)
```

Comment:
Purists before asked in my code review to use %w but idk if it worth it

Reply:
good call thanks